v2/libs/keystore/keystore.ts

```diff
@@ -183,16 +225,52 @@ export class Keystore {
 			type: 'internal',
 			label,
 			// @TODO: consider an MAC?
-			privKey: hexlify(aesCtr.encrypt(getBytes(privateKey))),
+			privKey: hexlify(aesCtr.encrypt(aes.utils.hex.toBytes(privateKey))),
 			meta: null
 		})
 		await this.storage.set('keystoreKeys', keys)
 	}
 
 	async removeKey(id: string) {
 		const keys: [StoredKey] = await this.storage.get('keystoreKeys', [])
-		if (!keys.find(x => x.id === id)) throw new Error(`keystore: trying to remove key that does not exist ${id}}`)
-		this.storage.set('keystoreKeys', keys.filter(x => x.id !== id))
+		if (!keys.find((x) => x.id === id)) throw new Error(`keystore: trying to remove key that does not exist ${id}}`)
+		this.storage.set(
+			'keystoreKeys',
+			keys.filter((x) => x.id !== id)
+		)
+	}
+
+	async getSigner(keyId: string) {
+		const storedKey: StoredKey = (await this.storage.get('keystoreKeys', [])).find((x: StoredKey) => x.id === keyId)
```

Comment:
even though i gave you the example, maybe this should be a two-liner, the first line doing the `await`, so that we don't have `(await ..)...`